### PR TITLE
Investigate to cancel a running save participant when saving again

### LINKED ISSUE #90509: Investigate to cancel a running save participant when saving again
It seems that if you save multiple times then waiting time for saving participants is not cancelled, probably due to not cancelling progress from 1st saving attempt. If it's not the case then this issue can be closed. Use case: After pressing save I spotted a typo and fixed it within few seconds and re-saved on PC with slow CPU.

### PR DESCRIPTION
**Pull request by bpasero**
@jrieken I was not so sure about this line: https://github.com/microsoft/vscode/compare/ben/90509?expand=1#diff-e434c7570ee4d27132355f82ecd18145L397

This PR fixes #90509

### CODE REVIEW

File: src/vs/workbench/api/browser/mainThreadSaveParticipant.ts [hunk 1]
```diff
@@ -394,7 +392,6 @@ export class SaveParticipant implements ISaveParticipant {
 			model.textEditorModel.pushStackElement();
 		}, () => {
 			// user cancel
-			cts.dispose(true);
 		});
```
Comment (jrieken) [inline src/vs/workbench/api/browser/mainThreadSaveParticipant.ts]: No more user-cancelling. Better leave the `const cts = new Cancellation...` and pass the `token`-parameter as ctor-arg

**Review by jrieken — CHANGES_REQUESTED** [review]

File: src/vs/workbench/api/browser/mainThreadSaveParticipant.ts (hunk 1 above)
Comment (bpasero) [reply src/vs/workbench/api/browser/mainThreadSaveParticipant.ts]: @jrieken got it, pushed

**Review by jrieken — APPROVED** [review]

**Comment by arhoy** [comment]
Running Save Participants keeps running forever for me! What do I do

**Comment by raptor235** [comment]
Started running into this today on php files... nothing has changed... no new extensions the files are the same and all of a sudden files stopped saving for me today...

**Pull request merged**